uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -64,13 +65,17 @@ define([
         }
       };
       vm.uiOptions = {
-        isAdmin: false,
         isChangeExpanded: false,
+        isManager: false, //when the dialog is opened by manager or admin
+        isOwner: false, //when the dialog is opened by the owner
         multipleDays: true,
+        contact: '',
         showDatePickerFrom: false,
         showDatePickerTo: false,
         userDateFormat: HR_settings.DATE_FORMAT,
+        selectedStatus: undefined,
         showBalance: false,
+        statusLabel: '',
```

Comment:
Why is this property even needed at all? You have access to the `leaveRequest.status_id` property, so in the view you can simply call one of those functions that you have in the component, that get you the OptionValue from the status_id. Something like
```html
<span class="col-xs-6">{{::$ctrl.getStatus($ctrl.leaveRequest.status_id).label}}</span>
```
This way you wouldn't have to deal with this additional property, and all the logic that it comes attached to